lib/Mojolicious/Plugin/Web/Auth/OAuth2.pm

```diff
@@ -116,7 +117,10 @@ sub _ua {
 
 sub _response_to_hash {
     my ( $self, $res ) = @_;
-    return ( $res->headers->content_type =~ /^application\/json[;]?/ )
+    my $types = Mojolicious::Types->new;
+    $types->type(json => ['application/json', 'text/javascript']);
+    my $exts = $types->detect( $res->headers->content_type );
+    return ( scalar(@$exts) && $exts->[0] eq 'json' )
```

Comment:
Does this line suffer from the parenthesis rule? I suspect it does.

Reply:
See https://perldoc.perl.org/perlop.html#Terms-and-List-Operators-(Leftward)

Reply:
16:57 <@mst> perlbot: deparse: return ( scalar(@$exts) && $exts->[0] eq 'json' )
16:57 < perlbot> mst: ( return ( scalar(@$exts) && ( $exts->[0] eq "json" ) ) 
                 );